### ISSUE TITLE
Impossibility to change the email of a user who is an actor of the ticket if the email is configured by default

### ISSUE THREAD

**Issue by amma35** [glpi-project/glpi#4080]
GLPI 9.3 & GLPI 9.2 & GLPI 9.1

Impossibility to change the email of a user who is an actor of the ticket if the email is configured by default 

* Steps to reproduce (which actions have you made) :
User with two email including "fisrt@mail.fr" by default :  
![user](https://user-images.githubusercontent.com/15138371/40620924-fa79ae3c-629a-11e8-8c7a-daf7ceb0fc6c.png)

Ticket with the user as the requester and selection of the address "second@mail.fr" :
![ticket](https://user-images.githubusercontent.com/15138371/40621003-51ece80a-629b-11e8-93f5-adf4b6d9c91f.png)

Changing the user's email address to "fisrt@mail.fr" : 
![change](https://user-images.githubusercontent.com/15138371/40621068-9da36f62-629b-11e8-9dc4-6e7c3e56334e.png)

* Actual result :
Error :
![error](https://user-images.githubusercontent.com/15138371/40621123-ca8bef36-629b-11e8-9eb0-af2864b148fc.png)

On the other hand, if I pass the address "second@mail.fr" by default : 
![user1](https://user-images.githubusercontent.com/15138371/40621282-6a7b8812-629c-11e8-9080-5592905828fc.png)

Same for users as an observer, assigned to.

Ticket with the user as the requester and selection of the address "second@mail.fr".
Changing the user's email address to "fisrt@mail.fr", it works.

**Comment by amma35** [comment]
After some searching, in the commonitilactor.class.php file, function showUserNotificationForm().

Displaying the dropdown list of email addresses, in the $emailtab table, the default email address has no key. 

```
foreach ($emails as $new_email) {
            if ($new_email != $default_email) {
               $emailtab[$new_email] = $new_email;
            } else {
              $emailtab[''] = $new_email;
            }
         }
```


So to the modification in the ticket_user.form.php file, the value 'alternative_email' is empty

**Comment by trasher** [comment]
Indeed, this condition seems a bit strange, and original commit message does not help. do you confirm your problem si solved with the following:
```diff
diff --git a/inc/commonitilactor.class.php b/inc/commonitilactor.class.php
index 1d06dc68d..c3d0a52dc 100644
--- a/inc/commonitilactor.class.php
+++ b/inc/commonitilactor.class.php
@@ -196,11 +196,7 @@ abstract class CommonITILActor extends CommonDBRelation {
          // Several emails : select in the list
          $emailtab = [];
          foreach ($emails as $new_email) {
-            if ($new_email != $default_email) {
-               $emailtab[$new_email] = $new_email;
-            } else {
-               $emailtab[''] = $new_email;
-            }
+            $emailtab[$new_email] = $new_email;
          }
          Dropdown::showFromArray("alternative_email", $emailtab,
                                  ['value'   => $this->fields['alternative_email']]);

 ```

**Comment by trasher** [comment]
Note: previous patch is to be applied on 9.2/bf branch ;)

**Comment by amma35** [comment]
Yes, the patch fixes the problem.

**Comment by trasher** [comment]
Thanks for the feedback
